homeassistant/components/compensation/__init__.py

```diff
@@ -78,8 +84,11 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
         degree = conf[CONF_DEGREE]
 
+        initial_coefficients: list[tuple[float, float]] = conf[CONF_DATAPOINTS]
+        sorted_coefficients = sorted(initial_coefficients, key=lambda x: x[0])
```

Comment:
Because if I add linear interpolation in the future, I don't want to sort the second via the second value, only the x.  It'll be a first come first serve at that point.